numba/interpreter.py

```diff
@@ -242,119 +243,169 @@ def _compute_generator_info(self):
 
     def _insert_var_dels(self):
         """
-        Insert ir.Del statements where necessary for the various
-        variables defined within the IR.
+        Insert del statements for each variable.
         """
-        # { var name -> { block offset -> stmt of last use } }
-        var_use = collections.defaultdict(dict)
-        cfg = self.cfa.graph
+        var_use_map, var_def_map = self._compute_use_defs()
+        live_map = self._compute_live_map(var_use_map, var_def_map)
+        dead_maps = self._compute_dead_maps(live_map, var_def_map)
+        intern_dead_map, escaping_dead_map = dead_maps
```

Comment:
Just for the record, you use "intern" in this PR but it should be "internal" ("intern" is a verb).